src/Microsoft.VisualStudio.Services.Agent/JobServer.cs

```diff
@@ -142,12 +153,46 @@ public async Task<BlobIdentifierWithBlocks> UploadLogToBlobstorageService(Stream
             // Since we read this while calculating the hash, the position needs to be reset before we send this 
             blob.Position = 0;
 
-            using(var blobClient = CreateArtifactsClient(_connection, default(CancellationToken)))
+            using (var blobClient = CreateArtifactsClient(_connection, default(CancellationToken)))
             {
                 return await blobClient.UploadBlocksForBlobAsync(blobId, blob, default(CancellationToken));
             }
         }
 
+        public async Task<(DedupIdentifier dedupId, ulong length)> UploadToBlobStore(bool verbose, string itemPath, CancellationToken cancellationToken)
```

Comment:
This seems similar to https://github.com/microsoft/azure-pipelines-agent/blob/afc5c2a1dd50632ba43cfb0afc646f3f4385d4ab/src/Agent.Worker/Build/FileContainerServer.cs#L323

Is there opportunity for reuse?

Reply:
Yes!